ts/runtime.ts

```diff
@@ -160,6 +153,79 @@ export class QuickJSRuntime implements Disposable {
     return context
   }
 
+  private getNewContextPointer(options: ContextOptions = {}): JSContextPointer {
+    let contextPointer
+    // If we are recycling an existing context, prioritize the existing pointer
+    if (options.contextPointer) {
+      contextPointer = options.contextPointer
+    } else if (options.intrinsics) {
+      contextPointer = this.ffi.QTS_NewContextRaw(this.rt.value)
+      this.addContextIntrinsics(contextPointer, options.intrinsics)
+    } else {
+      contextPointer = this.ffi.QTS_NewContext(this.rt.value)
+    }
+
+    return contextPointer
+  }
+
+  private addContextIntrinsics(ctxPtr: JSContextPointer, intrinsics: Intrinsic[]): void {
```

Comment:
Thanks for the PR. Can we avoid duplicating this s method? Why is it needed in both implementations?

Reply:
forsure. The method is the same across runtimes, I just forgot runtime-async extends runtime. Addressed in https://github.com/justjake/quickjs-emscripten/pull/116/commits/9836ad8e33b8020f5ee42d4d6fe00ba2953de614